resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -381,4 +382,12 @@ object KubernetesUtils extends Logging {
       }
     }
   }
+
+  @Since("3.3.0")
+  def loadFeatureStep(conf: KubernetesConf, className: String): KubernetesFeatureConfigStep = {
+    val feature =
+      Utils.classForName(className).newInstance().asInstanceOf[KubernetesFeatureConfigStep]
+    feature.init(conf)
```

Comment:
we need to call `KubernetesFeatureConfigStep.init(KubernetesConf)`

For driver, it is implemeted to call  `KubernetesDriverFeatureConfigStep.init(KubernetesDriverConf)`
For executor, it is implemeted to call  `KubernetesExecutorFeatureConfigStep.init(KubernetesExecutorConf)`